src/ariel-os-embassy-common/src/executor_swi.rs

```diff
@@ -18,15 +18,15 @@
 /// pub use interrupt::SWI_IRQ_1 as SWI;
 /// #[interrupt]
 /// unsafe fn SWI_IRQ_1() {
-///     unsafe { crate::EXECUTOR.on_interrupt() }
+///     unsafe { EXECUTOR.on_interrupt() }
 /// }
 /// ```
 ///
 /// Note: this expects the `interrupt` to be present (e.g., "used") and that it contains the ISR
 /// type.
 #[macro_export]
 macro_rules! executor_swi {
-    ($swi:ident) => {
+    ($exe:ident, $swi:ident) => {
```

Comment:
The type needs to be more generic, otherwise it wouldn't be possible to pass `crate::EXECUTOR` for instance; I think `expr` is the right one here:

```suggestion
    ($exe:expr, $swi:ident) => {
```